src/wallet/wallet.cpp

```diff
@@ -3112,6 +3119,30 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransac
     return true;
 }
 
+bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransactionRef& tx, CReserveKey& reservekey, CAmount& nFeeRet,
+                                int& nChangePosInOut, std::string& strFailReason, const CCoinControl& coin_control, bool sign)
+{
+    int nChangePosIn = nChangePosInOut;
+    CTransactionRef tx2 = tx;
+    bool res = _create_tx(vecSend, tx, reservekey, nFeeRet, nChangePosInOut, strFailReason, coin_control, sign);
+    // try with avoidpartialspends unless it's enabled already
+    if (res && !coin_control.m_avoid_partial_spends) {
+        CCoinControl tmp_cc = coin_control;
+        tmp_cc.m_avoid_partial_spends = true;
+        CAmount nFeeRet2;
+        int nChangePosInOut2 = nChangePosIn;
+        if (_create_tx(vecSend, tx2, reservekey, nFeeRet2, nChangePosInOut2, strFailReason, tmp_cc, sign)) {
```

Comment:
Can this branching for avoid_partial_spends on/off be done in `CWallet::SelectCoins` (where all different combinations are tried), instead of creating a second place of iterating through things?

Reply:
I would love to, but it relies on absolute fee to determine whether to force-use the avoid partial variant, which is not available in `SelectCoins`.